yt/frontends/swift/data_structures.py

```diff
@@ -0,0 +1,209 @@
+
+        return
+
+    def _get_info_attributes(self, dataset):
+        """
+        Gets the information from a header-style dataset and returns it as a
+        python dictionary.
+
+        Example: self._get_info_attributes(header) returns a dictionary of all
+        of the information in the Header.attrs.
+        """
+
+        with h5py.File(self.filename, "r") as handle:
+            header = dict(handle[dataset].attrs)
+
+        return header
+
+    def _parse_parameter_file(self):
+        """
+        Parse the SWIFT "parameter file" -- really this actually reads info
+        from the main HDF5 file as everything is replicated there and usually
+        parameterfiles are not transported.
+
+        The header information from the HDF5 file is stored in an un-parsed
+        format in self.parameters should users wish to use it.
+        """
+
+        self.unique_identifier = uuid4()
+
+        # Read from the HDF5 file, this gives us all the info we need. The rest
+        # of this function is just parsing.
+        header = self._get_info_attributes("Header")
+        runtime_parameters = self._get_info_attributes("RuntimePars")
+
+        policy = self._get_info_attributes("Policy")
+        # These are the parameterfile parameters from *.yml at runtime
+        parameters = self._get_info_attributes("Parameters")
+
+        # Not used in this function, but passed to parameters
+        hydro = self._get_info_attributes("HydroScheme")
+        subgrid = self._get_info_attributes("SubgridScheme")
+
+        self.domain_right_edge = header["BoxSize"]
+        self.domain_left_edge = np.zeros_like(self.domain_right_edge)
+
+        self.dimensionality = int(header["Dimension"])
+
+        # SWIFT is either all periodic, or not periodic at all
+        periodic = int(runtime_parameters["PeriodicBoundariesOn"])
+
+        if periodic:
+            self.periodicity = [True] * self.dimensionality
+        else:
+            self.periodicity = [False] * self.dimensionality
+
+        self.current_time = float(header["Time"])
+
+        # Now cosmology enters the fray, as a runtime parameter.
+        self.cosmological_simulation = int(policy["cosmological integration"])
+
+        if self.cosmological_simulation:
+            try:
+                self.current_redshift = float(header["Redshift"])
+                # These won't be present if self.cosmological_simulation is false
+                self.omega_lambda = float(parameters["Cosmology:Omega_lambda"])
+                self.omega_matter = float(parameters["Cosmology:Omega_m"])
+                # This is "little h"
+                self.hubble_constant = float(parameters["Cosmology:h"])
+            except KeyError:
+                mylog.info(
+                    "Could not find cosmology information in Parameters."
+                )
+                mylog.info(
+                    "Setting up as a non-cosmological run. Check this!"
```

Comment:
It would also be worth mentioning in this message that `policy["cosmological integration"]` is a non-null value so swift is saying it was configured to run with comoving coordinates but no cosmology parameters are present, which would be an odd inconsistent state. I might also change these to `mylog.warn` or even a user-visible runtime warning to more loudly notify them that their dataset is odd or possibly broken. Unless of course this can happen a lot.

Reply:
Yep - I'm fairly sure if this happens something really dodgy is going on!

Reply:
I am a little confused by what you mean Nathan. In SWIFT, cosmological integration is what we call an "engine policy"; you do not have to compile with(out) cosmological integration. To run with cosmology, you just pass an extra flag to the binary (`swift -c ...`). If you attempt to run with `-c` and you do not have the cosmological parameters in your parameter file, we don't let you run.

Reply:
the issue here is the file _thought_ it was a cosmo run as `self.cosmological_simulation` is true, but then when we tried to load the parameters we failed. Surely, that is a strange case?

Reply:
Ah yeah, that is a very strange case; hence the logging information. Perhaps it should be expanded to say "Could not find cosmology information in Parameters, despite having ran with -c signifying a cosmological run."